dbt_common/artifacts/catalog.py

```diff
@@ -0,0 +1,26 @@
+from dataclasses import dataclass
```

Comment:
Could we move this file into a dbt_common/contracts/metadata.py? We have something similar in place for dbt_common_contracts/constraints.py: https://github.com/dbt-labs/dbt-common/tree/main/dbt_common/contracts

'artifacts' and 'catalog' are more core concerns and don't semantically mean much in the context of dbt-common which is much more generic.